### PR TITLE
Add nixos tests with names starting by A or B to all relevant packages

### PR DESCRIPTION
**Pull request by Ekleog**
`blockbook` already failed to build before this change, AFAICT all other packages build both before and after.

### CODE REVIEW

**Review by SuperSandro2000 — COMMENTED** [review]
I don't know if it is a good idea to use tests from other programs. This could lead to a lot of false positives and cross effects.

**Comment by Ekleog** [comment]
That's definitely true, OTOH AFAICT right now the tooling is using these mostly to inform the user — I'd think that a possibility for false positives would be better than no test at all in such a situation, especially if the nixos test explicitly has the name of the other program to make it clear that we're testing integration between two programs and thus that the problem could stem from the other program :)

That being said, a few of these bother me a bit, like using the borgbackup (little-used program) test for openssh (much-used program)… but OTOH we currently have no passthru.tests for openssh, so having something with possibly some FP if borgbackup is currently broken on master is already better than nothing.

**Especially with tooling that (like nixpkgs-check, and maybe we can backport that to other toolings?) checks whether a breakage is new or was already present on master and errors only upon confirmed regression :)** (emphasis because it's probably the most important part of the message, once we get our tooling to error only on regression and not just on non-build we're good to add as many false positives as we want so long as they're stable)

Over all, my thoughts since introducing it are “passthru.tests should be a hand-picked reasonably fast to build subset of what nixpkgs-review would build, that exerts as much as possible of the API exposed by the package while having as little false positives as possible” — and with that in mind I'm thinking that these additions make sense.

Particularly given that the reciprocal is also true: a breakage in openssh would break the borgbackup test and thus cause false positives there — it's just the way integration testing works.

WDYT?

File: pkgs/applications/blockchains/bitcoin.nix [hunk 1]
```diff
@@ -81,6 +82,9 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
+  passthru.tests = {
+    smoke-test = nixosTests.bitcoind;
+  };
```
Comment (davidak) [inline pkgs/applications/blockchains/bitcoin.nix:87]: @Ekleog can you add a newline after this block? in all files

**Comment by Ekleog** [comment]
Okay, I've handled the review comment, rebased and ran nixpkgs-check. If any of the “integration” tests that don't test exclusively the related package adds a false positive that end up being a bother in any situation, please feel free to delete it! I have some hope that in practice they will bring more benefits than problems, but I may totally be wrong.

For now, given the result of nixpkgs-check below, I'm going to land this :)

### nixpkgs-check report

**version:** `nixpkgs-check v0.1.0` on NixOS 21.05, sandbox = "true"
**packages declared changed:** {"beanstalkd", "bcachefs-tools", "miniupnpd", "git", "blockbook", "bees", "buildkite-agent", "opentracker", "nginx", "ocamlPackages.atd", "buildbot-worker", "bitcoind", "pebble", "avahi", "openssh", "transmission", "pythonPackages.selenium", "_3proxy", "apacheHttpd"}
**manual tests declared performed:**
 * 😢 not built on NixOS
 * 😢 not built on MacOS
 * 😢 not built on Other Linux distributions

**complies with contributing.md:** ✔ yes
**package beanstalkd:** ✔ continued building
**package bcachefs-tools:** ✔ continued building
**package miniupnpd:** ✔ continued building
**package git:** ✔ continued building
**package blockbook:** 😢 still does not build
**package bees:** ✔ continued building
**package buildkite-agent:** ✔ continued building
**package opentracker:** ✔ continued building
**package nginx:** ✔ continued building
**package ocamlPackages.atd:** ✔ continued building
**package buildbot-worker:** ✔ continued building
**package bitcoind:** ✔ continued building
**package pebble:** ✔ continued building
**package avahi:** ✔ continued building
**package openssh:** ✔ continued building
**package transmission:** ✔ continued building
**package pythonPackages.selenium:** ✔ continued building
**package _3proxy:** ✔ continued building
**package apacheHttpd:** ✔ continued building
**closure size for beanstalkd:** ✔ stayed constant, at 33.0 MB
**tests of beanstalkd:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of beanstalkd:**
  * *updated binaries:*
    * 😢 beanstalkd was not run


**closure size for bcachefs-tools:** ✔ stayed constant, at 154.4 MB
**tests of bcachefs-tools:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of bcachefs-tools:**
  * *updated binaries:*
    * 😢 mount.bcachefs.sh was not run
    * 😢 mkfs.bcachefs was not run
    * 😢 fsck.bcachefs was not run
    * 😢 bcachefs was not run


**closure size for miniupnpd:** ✔ stayed constant, at 59.2 MB
**tests of miniupnpd:**
  * *added tests:*
    * 💚 bittorrent-integration was run successfully


**binaries of miniupnpd:**
  * *updated binaries:*
    * 😢 miniupnpd was not run


**closure size for git:** ✔ stayed constant, at 284.3 MB
**tests of git:**
  * *added tests:*
    * 💚 buildbot-integration was run successfully


**binaries of git:**
  * *updated binaries:*
    * 😢 git-http-backend was not run
    * 😢 git-shell was not run
    * 😢 git was not run
    * 😢 git-receive-pack was not run
    * 😢 git-upload-pack was not run
    * 😢 git-credential-netrc was not run
    * 😢 git-cvsserver was not run
    * 😢 git-upload-archive was not run


**closure size for bees:** ✔ stayed constant, at 180.6 MB
**tests of bees:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of bees:**
  * *updated binaries:*
    * 😢 bees-service-wrapper was not run
    * 😢 beesd was not run


**closure size for buildkite-agent:** ✔ stayed constant, at 304.7 MB
**tests of buildkite-agent:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of buildkite-agent:**
  * *updated binaries:*
    * 😢 buildkite-agent was not run


**closure size for opentracker:** ✔ stayed constant, at 33.2 MB
**tests of opentracker:**
  * *added tests:*
    * 💚 bittorrent-integration was run successfully


**binaries of opentracker:**
  * *updated binaries:*
    * 😢 opentracker was not run


**closure size for nginx:** ✔ stayed constant, at 60.1 MB
**tests of nginx:**
  * *added tests:*
    * 💚 acme-integration was run successfully

  * *updated tests:*
    * ✔ nginx-auth continued running successfully
    * ✔ nginx-etag continued running successfully
    * ✔ nginx-sandbox continued running successfully
    * ✔ nginx-pubhtml continued running successfully
    * ✔ nginx-sso continued running successfully
    * ✔ variants continued running successfully
    * ✔ nginx continued running successfully


**binaries of nginx:**
  * *updated binaries:*
    * 😢 nginx was not run


**closure size for ocamlPackages.atd:** ✔ stayed constant, at 474.7 MB
**tests of ocamlPackages.atd:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of ocamlPackages.atd:**
  * *updated binaries:*
    * 😢 atdcat was not run


**closure size for buildbot-worker:** ✔ stayed constant, at 135.9 MB
**tests of buildbot-worker:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of buildbot-worker:**
  * *updated binaries:*
    * 😢 buildbot_worker_windows_service was not run
    * 😢 buildbot-worker was not run


**closure size for bitcoind:** ✔ stayed constant, at 139.3 MB
**tests of bitcoind:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of bitcoind:**
  * *updated binaries:*
    * 😢 test_bitcoin was not run
    * 😢 bitcoind was not run
    * 😢 bitcoin-wallet was not run
    * 😢 bitcoin-cli was not run
    * 😢 bitcoin-tx was not run


**closure size for pebble:** ✔ stayed constant, at 49.8 MB
**tests of pebble:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of pebble:**
  * *updated binaries:*
    * 😢 pebble-challtestsrv was not run
    * 😢 pebble was not run


**closure size for avahi:** ✔ stayed constant, at 173.0 MB
**tests of avahi:**
  * *added tests:*
    * 💚 smoke-test-resolved was run successfully
    * 💚 smoke-test was run successfully


**binaries of avahi:**
  * *updated binaries:*
    * 😢 avahi-browse-domains was not run
    * 😢 avahi-resolve was not run
    * 😢 avahi-publish-address was not run
    * 😢 avahi-set-host-name was not run
    * 😢 avahi-publish was not run
    * 😢 avahi-resolve-host-name was not run
    * 😢 avahi-dnsconfd was not run
    * 😢 avahi-browse was not run
    * 😢 avahi-autoipd was not run
    * 😢 avahi-daemon was not run
    * 😢 avahi-publish-service was not run
    * 😢 avahi-resolve-address was not run


**closure size for openssh:** ✔ stayed constant, at 102.0 MB
**tests of openssh:**
  * *added tests:*
    * 💚 borgbackup-integration was run successfully


**binaries of openssh:**
  * *updated binaries:*
    * 😢 ssh-keygen was not run
    * 😢 sshd was not run
    * 😢 ssh-keyscan was not run
    * 😢 ssh-copy-id was not run
    * 😢 ssh was not run
    * 😢 ssh-agent was not run
    * 😢 sftp was not run
    * 😢 ssh-add was not run
    * 😢 scp was not run


**closure size for transmission:** ✔ stayed constant, at 156.6 MB
**tests of transmission:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of transmission:**
  * *updated binaries:*
    * 😢 transmission-edit was not run
    * 😢 transmission-cli was not run
    * 😢 transmission-show was not run
    * 😢 transmission-daemon was not run
    * 😢 transmission-create was not run
    * 😢 transmission-remote was not run


**closure size for pythonPackages.selenium:** ✔ stayed constant, at 115.9 MB
**tests of pythonPackages.selenium:**
  * *added tests:*
    * 💚 testing-bitwarden was run successfully



**closure size for _3proxy:** ✔ stayed constant, at 35.1 MB
**tests of _3proxy:**
  * *added tests:*
    * 💚 smoke-test was run successfully


**binaries of _3proxy:**
  * *updated binaries:*
    * 😢 mycrypt was not run
    * 😢 ftppr was not run
    * 😢 tcppm was not run
    * 😢 3proxy was not run
    * 😢 proxy was not run
    * 😢 udppm was not run
    * 😢 pop3p was not run
    * 😢 socks was not run


**closure size for apacheHttpd:** ✔ stayed constant, at 132.9 MB
**tests of apacheHttpd:**
  * *added tests:*
    * 💚 acme-integration was run successfully


**binaries of apacheHttpd:**
  * *updated binaries:*
    * 😢 rotatelogs was not run
    * 😢 dbmmanage was not run
    * 😢 htpasswd was not run
    * 😢 envvars-std was not run
    * 😢 htdigest was not run
    * 😢 fcgistarter was not run
    * 😢 envvars was not run
    * 😢 httpd was not run
    * 😢 htdbm was not run
    * 😢 checkgid was not run
    * 😢 htcacheclean was not run
    * 😢 logresolve was not run
    * 😢 httxt2dbm was not run
    * 😢 ab was not run
    * 😢 apachectl was not run

**Pull request merged**